### PR TITLE
modify zap v11 to conform to new plugin interface

### PR DESCRIPTION
**Pull request by mschoch**

### CODE REVIEW

**Comment by mschoch** [comment]
Not sure why, it won't let me add @sreekanth-cb as a reviewer.

While the content of this is sound, the go.mod file is a bit half-baked as not enough of the pieces are in place yet for me to really verify it.  For the same reason, there is no go.sum yet, because the go.mod is hand edited.

File: merge.go [hunk 1]
```diff
@@ -31,32 +31,31 @@ import (
...
-// Caller should provide EITHER segments or memSegments, but not both.
-// This API is experimental and may be removed at any time.
-var ValidateMerge = func(segments []*Segment, memSegments []*SegmentBase, drops []*roaring.Bitmap, newSegment *Segment) error {
-	return nil
-}
-
 const docDropped = math.MaxUint64 // sentinel docNum to represent a deleted doc
 
-// Merge takes a slice of zap segments and bit masks describing which
+// Merge takes a slice of segments and bit masks describing which
```
Comment (sreekanth-cb) [inline merge.go:36]: May be need to clean up the comments section on the chunkFactor, as it no longer takes a chunk factor?

File: posting.go [hunk 2]
```diff
@@ -793,10 +793,23 @@ func (p *PostingsIterator) DocNum1Hit() (uint64, bool) {
...
 }
 
+// ActualBitmap returns the underlying actual bitmap
+// which can be used up the stack for optimizations
+func (p *PostingsIterator) ActualBitmap() *roaring.Bitmap {
+	return p.ActualBM
+}
+
+// ReplaceActual replaces the ActualBM with the provided
+// bitmap AND also
```
Comment (sreekanth-cb) [inline posting.go]: AND also ...?

File: plugin.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package zap
+
+import (
+	"github.com/blevesearch/bleve/index/scorch/segment"
+)
+
+type ZapPlugin struct{}
```
Comment (sreekanth-cb) [inline plugin.go:23]: Many a times while reading code, I find it difficult to track/load into mind about 
-whether the current code is an implementation of any interface Or if so, where can I find the interface details.. 
Not sure of the right of way of doing this.. But having a comment here referring to the segment Plugin interface would be desirable. Also understand that the plugin code isn't there yet in blevesearch/bleve..
Some way to easily trace back the connection bw interface & implementation is nice..

File: merge.go [hunk 4]
```diff
@@ -31,32 +31,31 @@ import (
...
 	segmentBases := make([]*SegmentBase, len(segments))
 	for segmenti, segment := range segments {
-		segmentBases[segmenti] = &segment.SegmentBase
+		switch segmentx := segment.(type) {
+		case *Segment:
+			segmentBases[segmenti] = &segmentx.SegmentBase
+		case *SegmentBase:
+			segmentBases[segmenti] = segmentx
+		default:
+			panic(fmt.Sprintf("oops, unexpected segment type: %T", segment))
```
Comment (sreekanth-cb) [inline merge.go:51]: type is segment vs segmentx?

Comment (mschoch) [reply merge.go:51]: I don't understand the question.  The name `segment` refers to the current segment in the for loop.  The name `segmentx` is the underlying implementation of that segment after type assertion.  Are you suggesting a different name?

File: merge.go (hunk 1 above)
Comment (mschoch) [reply merge.go:36]: Tried to address in second commit.

File: posting.go (hunk 2 above)
Comment (mschoch) [reply posting.go]: Tried to address in second commit.

File: plugin.go (hunk 3 above)
Comment (mschoch) [reply plugin.go:23]: Tried to address in second commit.

File: merge.go (hunk 4 above)
Comment (sreekanth-cb) [reply merge.go:51]: nm. I thought it had to be segmentx for printing the type. But now I realise that %T would do the trick.

**Review by sreekanth-cb — APPROVED** [review]
lgtm +2.

**Pull request merged**